### PR TITLE
Promote PodProcessNamespaceSharing feature to GA

### PR DESCRIPTION
**Pull request by verb**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**: This promotes the PodProcessNamespaceSharing from beta to GA

The [Process Namespace Sharing KEP](https://git.k8s.io/enhancements/keps/sig-node/20190920-pod-pid-namespace.md) specifies the following criteria for Beta -> GA graduation:

- [x] 3 articles describing using `shareProcessNamespace` in a task.
  - [The Almight Pause Container](https://www.ianlewis.org/en/almighty-pause-container) blog post by Ian Lewis; 10 Oct 2017
  - [DevOps with Kubernetes](https://books.google.ch/books?id=eCWGDwAAQBAJ&pg=PA327&lpg=PA327&dq=Kubernetes+shareProcessNamespace&source=bl&ots=QxUcZiL4ZG&sig=ACfU3U24Agyc2lJ57l6OiVbZNrXB61Rm3A&hl=en&sa=X&ved=2ahUKEwiUnLnetrflAhUa6aYKHTTSAHA4FBDoATABegQICRAB#v=onepage&q=Kubernetes%20shareProcessNamespace&f=false) by Hideto Saito, Hui-Chuan Chloe Lee, Cheng-Yang Wu; Packt Publishing; 31 Jan 2019
  - [Kubernetes Tips and Tricks](https://thepracticalsysadmin.com/kubernetes-tips-and-tricks/) by Josh Reichardt; Practical System Administration; 18 December 2018
- [x] `shareProcessNamespace` referenced in documentation for a Kubernetes cloud
  provider (as proxy indicator for feature enabled on cloud provider).
  - Google Cloud: [Running a Cloud MQTT bridge in Kubernetes](https://cloud.google.com/community/tutorials/kube-mqtt)
- [x] Feature spends at least 2 release in beta.
- [x] No open bug reports for latest release.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Pod process namespace sharing is now Generally Available. The `PodShareProcessNamespace` feature gate is now deprecated and will be removed in Kubernetes 1.19.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
- [KEP]: https://git.k8s.io/enhancements/keps/sig-node/20190920-pod-pid-namespace.md
- [Usage]: https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -519,7 +521,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	LocalStorageCapacityIsolation:  {Default: true, PreRelease: featuregate.Beta},
 	Sysctls:                        {Default: true, PreRelease: featuregate.Beta},
 	EphemeralContainers:            {Default: false, PreRelease: featuregate.Alpha},
-	PodShareProcessNamespace:       {Default: true, PreRelease: featuregate.Beta},
+	PodShareProcessNamespace:       {Default: true, PreRelease: featuregate.GA},                      // remove in 1.19
```
[the remark this answers is not in the document]
Comment (liggitt) [reply pkg/features/kube_features.go]: and remove checks of the feature gate in code (and any code that only executed when the feature gate was disabled)

[the remark this answers is not in the document]
Comment (verb) [reply pkg/features/kube_features.go]: Ah, makes sense. I opened #84638 to add instructions for this.

**Comment by liggitt** [comment]
mechanics look good, let me know once this has SIG review/approval

**Comment by yujuhong** [comment]
/lgtm
/approve

Adding sig-node leads for approval: @dchen1107 @derekwaynecarr

**Comment by verb** [comment]
/assign @liggitt 

Hi Jordan, this now has sig-node approval and is ready for 1.17. Thanks!

**Comment by verb** [comment]
Looks like the approve didn't take. Testing:

/approve

**Pull request merged**